src/components/AddItems.js

```diff
@@ -36,6 +42,24 @@ const AddItemsToList = () => {
       .get()
       .then((data) => {
         if (data.docs.length) {
+          const { items } = data.docs[0].data();
+          const nonWordCharactersAndUnderscores = /[\W_]/g;
+          const shoppingListItemExists = items.some(
+            (shoppingListItemObject) => {
+              return (
+                shoppingListItemObject.shoppingListItemName
+                  .toLowerCase()
+                  .replace(nonWordCharactersAndUnderscores, '') ===
+                shoppingListItemName
+                  .toLowerCase()
+                  .replace(nonWordCharactersAndUnderscores, '')
```

Comment:
Here is some duplicate code that's a great candidate for being in a seperate function:
You can make a `const normalizeString = str => str.toLowerCase().replace(nonWordCharactersAndUnderscores, '');` or similar function that takes a string and returns it's normalized version. This can make the code more readable